src/BUILD_WINDOWS.md

```diff
@@ -40,7 +40,43 @@ If using anything else other than Visual Studio 2008 for development, your code
 
 It is OK to add newer Visual Studio (2015, 2017) solution files to the project, but there then must be dual solution files for both Visual C++ 2008 and the latest Visual Studio.
 
-Note: There is an update to the CMake configuration that adds support for Windows in the works for future use.
+Build and Development Instructions with Visual Studio 2017 & CMake
+---------------------------------------------------------------------
+
+An alternative method for development of the SoftEtherVPN project on Windows is through CMake.
+
+There are several methods for using CMake but the easiest by far is through Visual Studio 2017 by importing the CMake project directly
+into it. So that is what will be described below.
+
+Requirements:
+
+1. Download Visual Studio 2017 (Community Edition is fine).
+2. During install, make sure to check "Desktop development with C++" under "Workloads".
+3. Click on individual components and scroll until you see "Visual C++ tools for CMake" under the compilers section. Make sure this is checked.
+4. Continue scrolling down until you reach "SDKs, libraries, and frameworks". Under this tab you want to check to options:
+    1. Visual C++ ATL for x86 and x64
+    2. Visual C++ MFC for x86 and x64
+
+5. Proceed with and finish Visual Studio 2017 install.
+6. You will need the `OpenSSL` and `libcrypto` headers to satisfy CMake. Both can be installed using the community provided Windows installer for
+the OpenSSL project. Install and leave everything on the default to assure that itdoes not install the headers to OpenSSL's `/bin` path, which is a possible install option (Not the default).
+7. Install the needed submodules to build the project, avoiding CMake telling you to do so with: `git submodule update --init --recursive`
+
+Building:
+
+Once both installs have finished, launch Visual Studio. Once its started go to the File menu click `Open --> CMake`. Then navigate to where you
+cloned the project and open the `CMakeLists.txt` file in the projects root directory.
+
+Visual Studio will proceed to start the CMake configuration process and once its finished, you can simply go to toolbar and click `CMake -> Build All`.
+
+Once it has finished, hopefully with no errors, look in the newly created `/build` directory in the project's folder. Inside are the development versions
+of all the SoftEtherVPN components.
+
+Congrats, you now have a complete CMake development environment for SoftEtherVPN on Windows, enjoy and happy contributing!
+
+Download Links:
+- (One of a few) OpenSSL Windows Installer: https://slproweb.com/products/Win32OpenSSL.html
```

Comment:
can we use nuget here ?

Reply:
I don't think so. Nuget is for C#/.NET projects. Everything I read says you need to download and have the openssl libs somewhere that your build can find them.

Reply:
Installing OpenSSL is not required, as we provide the static libraries in the repository:
https://github.com/SoftEtherVPN/SoftEtherVPN/blob/080edcdb8155ebaf110dfab1bd747a7fd0728203/src/Mayaqua/CMakeLists.txt#L25-L45

Reply:
@davidebeatrici Is there a way to tell CMake this by default? It didn't pick them up automatically.

Reply:
It should find the libraries automatically, as we specify the path inside the repository.

Reply:
Odd. Removed the ones I had separately and it still worked. I'll just put it up to my system being weird. Incoming commit removing that.

Reply:
No problem.

Reply:
Should be fine now. CI passed for the changes removing the OpenSSL info

Reply:
Could you squash the commits, please?

Reply:
@davidebeatrici So far I have made 3 messes out of my repo. Can you link me to a good way to squash commits?

Reply:
Sure: https://stackoverflow.com/a/5189600

Reply:
I think that worked?

Reply:
Indeed!